cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaMetadataStateManagerTest.java

```diff
@@ -0,0 +1,212 @@
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.api.kafka.model.kafka.Kafka;
+import io.strimzi.api.kafka.model.kafka.KafkaBuilder;
+import io.strimzi.api.kafka.model.kafka.KafkaMetadataState;
+import io.strimzi.api.kafka.model.kafka.KafkaStatus;
+import io.strimzi.api.kafka.model.kafka.listener.GenericKafkaListenerBuilder;
+import io.strimzi.api.kafka.model.kafka.listener.KafkaListenerType;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.operator.common.Reconciliation;
+
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaft;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftDualWriting;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftPostMigration;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.PreKRaft;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.KRaftMigration;
+import static io.strimzi.api.kafka.model.kafka.KafkaMetadataState.ZooKeeper;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class KafkaMetadataStateManagerTest {
+
+    private static final String CLUSTER_NAMESPACE = "my-namespace";
+
+    private static final String CLUSTER_NAME = "kafka-test-cluster";
+
+    private static final Reconciliation RECONCILIATION = new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, CLUSTER_NAMESPACE, CLUSTER_NAME);
+
+    private static final Kafka KAFKA = new KafkaBuilder()
+            .withNewMetadata()
+            .withName(CLUSTER_NAME)
+            .withAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled"))
+            .withNamespace(CLUSTER_NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+            .withNewKafka()
+            .withListeners(new GenericKafkaListenerBuilder()
+                    .withName("plain")
+                    .withPort(9092)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(false)
+                    .build())
+            .endKafka()
+            .endSpec()
+            .build();
+
+    private KafkaMetadataState checkTransition(Kafka kafka, KafkaStatus status, boolean useBooleanKraftFeatureEnabled) {
+
+        KafkaMetadataStateManager kafkaMetadataStateManager = new KafkaMetadataStateManager(RECONCILIATION, kafka, useBooleanKraftFeatureEnabled);
+
+        return kafkaMetadataStateManager.computeNextMetadataState(status);
+    }
+
+    @Test
+    public void testMoveFromZookeeperToKRaftMigrationWhenMigrationAnnotationEnabled() {
+
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .withNewStatus()
+                .withKafkaMetadataState("ZooKeeper")
+                .endStatus()
+                        .build();
+
+        assertEquals(kafka.getStatus().getKafkaMetadataState(), "ZooKeeper");
```

Comment:
what are you testing here? I think it just tests that the `KafkaBuilder` class works :-)
You use it to add a metadata state and then you check on the `Kafka` instance that the state is there. It's not testing the metadata state manager at all. We can remove this.